### PR TITLE
monitoring with prometheus and grafana

### PR DESCRIPTION
**Pull request by joshuafernandes**
## PR description
- push metrics to prometheus via the prometheus gateway
- grafana dashboard setup to import the prometheus datasource and loads up the [Pantheon dashboard](https://grafana.com/dashboards/10273)

## Fixed Issue(s)
fixes IT-26

### CODE REVIEW

File: docker-compose-prometheus-scraper.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
```
Comment (joshuafernandes) [inline docker-compose-prometheus-scraper.yml]: Added an example + config (monitoring/prometheus) for use with the prometheus scraper

File: .env [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 # the version of Pantheon Docker image to use.
 # see https://hub.docker.com/r/pegasyseng/pantheon/tags for possible values
-PANTHEON_VERSION=1.1.1
+PANTHEON_VERSION=1.1.2
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply .env]: Sure thing, I'll revert this change

File: list.sh [hunk 3]
```diff
@@ -50,5 +50,7 @@ else
   echo "JSON-RPC WebSocket service endpoint : ws://${HOST}:${explorerMapping##*:}/jsonws"
   echo "GraphQL HTTP service endpoint       : http://${HOST}:${explorerMapping##*:}/graphql"
   echo "Web block explorer address          : http://${HOST}:${explorerMapping##*:}"
+  echo "Prometheus address                  : http://${HOST}:9090"
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply list.sh]: Yup, its exposed as a volume too so they can mount to somewhere local and examine metrics later if they want. Is this un-intended behaviour?

[the remark this answers is not in the document]
Comment (ajsutton) [reply list.sh]: Prometheus has its own UI with ability to visualise and explore the available metrics as well as support for alerts.  Its not as pretty as Grafana but it is very useful to be able to get to it for some things.

File: docker-compose.yml [hunk 4]
```diff
@@ -50,12 +71,50 @@ services:
...
+    command:
+      - --config.file=/etc/prometheus/prometheus.yml
+      - --web.external-url=http://localhost:9090/prometheus
+      - --web.route-prefix=/prometheus
+
+  grafana:
+    image: "grafana/grafana"
+    environment:
+      - GF_SERVER_ROOT_URL=%(protocol)s://%(domain)s/grafana
+      - GF_AUTH_ANONYMOUS_ENABLED=true
```
Comment (joshuafernandes) [inline docker-compose.yml]: @NicolasMassart no password anymore

**Comment by joshuafernandes** [comment]
@NicolasMassart I've updated to use the pull model (prometheus scrapes the nodes it knows about) and it looks like this now

Prometheus
image

Grafana:
image

**Pull request merged**